File: functions/financial_calc_update.py
# Josephine Chiu, Financial Calculator Update for Python

# Josephine Chiu, Financial Calculator Update for Python
print("Hello! This is your personal budget calculator! Please follow the instructions to learn more about your monthly finances.")

def user(type):
    question = "How much is your monthly " + type + "?\n"
    return float(input(question))

income = user("income")
rent = user("rent")
utilities = user("utilities")
groceries = user("groceries")
transportation = user("transportation")

savings = income*0.1
spending = rent+utilities+groceries+transportation

def percent(income, amount, type):
    percent_type = float(amount/income*100)
    print(f"You spent ${amount:.2f} on {type} and that is {percent_type:.0f}% of your income")

percent(income, rent, "rent")
percent(income, utilities, "utilities")
percent(income, groceries, "groceries")
percent(income, transportation, "transportation")
percent(income, spending, "spending")

print(f"You saved ${savings:.2f} and that is 10% of your income")

File: functions/financial_calc_update.c
// Josephine Chiu, Financial Calculator Update for C 

#include <stdio.h>

float input(char*type, float*input){
    printf("What is your monthly %s:\n", type);
    scanf("%f", input);
    return *input;
}

void percent(float cost, float income, const char* type) {  
    float percent = (cost / income) * 100;  
    printf("Your %s is $%.2f, which is %.2f%% of your income.\n", type, cost, percent);  
}
float total;
float spendings;
float savings;
float rent;
float income;
float utilities;
float groceries;
float transportation;

int main(void){

printf("Welcome to the Financial Calculator! Please answer the follwing questions in order to fully analyze your savings and expenses.\n");

income= input("income", &income);
rent = input("rent", &rent);
utilities = input("utilities", &utilities);
groceries = input("groceries", &groceries);
transportation = input("transportation", &transportation);

savings = income*0.1;
spendings = income-savings-rent-utilities-groceries-transportation;
total = income - savings - spendings;

percent(total, income, "total");
percent(rent, income, "rent");  
percent(utilities, income, "utilities");  
percent(groceries, income, "groceries");  
percent(transportation, income, "transportation");  
percent(savings, income, "savings");  
percent(spendings, income, "spendings");  
    return 0;
}

//hi Ms. larose. This is josephine. I tried my best. :)